gravatar/src/main/java/com/gravatar/di/container/GravatarSdkContainer.kt

```diff
@@ -20,8 +19,6 @@ internal class GravatarSdkContainer private constructor() {
 
     private fun getRetrofitBaseBuilder() = Retrofit.Builder().baseUrl(GRAVATAR_BASE_URL)
 
-    val dispatcherDefault = Dispatchers.Default
-    val dispatcherIO = Dispatchers.IO
```

Comment:
From the document you linked:

> This dependency injection pattern makes testing easier as you can replace those dispatchers in unit and instrumentation tests with a [test dispatcher](https://developer.android.com/kotlin/coroutines/coroutines-best-practices#test-coroutine-dispatcher) to make your tests more deterministic.

Injecting dispatchers is a good practice for making testing easier, but using `Dispatchers.setMain` seems even easier to me. I wonder if they're other reasons though.

Reply:
Just adding my 2 cents: Ignore if they don't help in anyway :)

> Injecting dispatchers is a good practice for making testing easier, but using Dispatchers.setMain seems even easier to me. I wonder if they're other reasons though.

I believe it's nice to have a central place to configure dispatchers. Perhaps in the event you might need a custom dispatcher. Probably not relevant to this project to define custom dispatchers but generally its nice to have a central place to do this kind of thing.

Also, as stated in the linked documentation, having your dispatchers centrally managed can make it easier to swap them out with a test dispatcher thus making the test more deterministic - Yeah, I know nothing new here, just wanted to rehash it again :)

Reply:
> Injecting dispatchers is a good practice for making testing easier, but using Dispatchers.setMain seems even easier to me. I wonder if they're other reasons though.

Besides the nice central place for configuration like @eyedol mentioned, `setMain` only replaces the `Main` dispatcher and the function exists because many built-in coroutine functions and helpers use the main dispatcher implicitly (so we are not able to inject `testDispatcher` for example). 

If you are using other Dispatchers (such as `IO` and `Default`) `setMain` doesn't help you and you need a way of replacing it with a different dispatcher, mostly for testing, since using background threads during tests can lead to wrong behavior or just add unnecessary difficulty. That's where injecting via constructor or even DI frameworks are useful.